frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/YarnMojo.java

```diff
@@ -60,15 +71,23 @@ protected boolean skipExecution() {
 
     @Override
     public synchronized void execute(FrontendPluginFactory factory) throws TaskRunnerException {
-        File packageJson = new File(this.workingDirectory, "package.json");
-        if (this.buildContext == null || this.buildContext.hasDelta(packageJson)
-            || !this.buildContext.isIncremental()) {
-            ProxyConfig proxyConfig = getProxyConfig();
-            boolean isYarnBerry = isYarnrcYamlFilePresent(this.session, this.workingDirectory);
-            factory.getYarnRunner(proxyConfig, getRegistryUrl(), isYarnBerry).execute(this.arguments,
-                this.environmentVariables);
+        IncrementalMojoHelper incrementalHelper = new IncrementalMojoHelper(incremental, workingDirectory, getLog());
+
+        if (incrementalHelper.shouldExecute()) {
+            File packageJson = new File(this.workingDirectory, "package.json");
+            if (this.buildContext == null || this.buildContext.hasDelta(packageJson)
+                    || !this.buildContext.isIncremental()) {
```

Comment:
Do I understand this correctly?

- we initialize Incremental helper (L74)
- we ask it to check the project (L76)
- if it reports yes, we check whether the incremental builds are enabled in the build context (L79)

That seems illogical to me, as the incremental setting on the build should in my mental picture should have primacy over the incremental helper. What am I missing?

Reply:
Yes - our helper has "priority" because we were not able to make the incremental setting on the build context to ever be true - any ideas how to achieve this are welcome (ie. in all my testing the buildcontext was the "default build context" that has a hard-coded implementation of all such methods combing down to "return false")

Please note the helper is explicitly enabled by the user in the pom.xml via the incremental flag so it's 100% clear what "happens" and that the build will run with these additional checks.